src/services/addonSettings.js

```diff
@@ -22,11 +22,22 @@ const apiSuffix = (projectName, siteId, variant, asVersion) => {
 const addonSettings = ayonApi.injectEndpoints({
   endpoints: (build) => ({
     getAddonSettingsList: build.query({
-      query: ({ variant, projectName, siteId }) => ({
-        url: `/api/settings`,
-        method: 'GET',
-        params: { variant, project_name: projectName, site_id: siteId, summary: true },
-      }),
+      query: ({ variant, projectName, siteId }) => {
+
+        // this should prevent passing null/undfiend values to the query
+        // params once for all (until we have typescript)
+        const params = {}
+        if (variant) params.variant = variant
+        if (projectName) params.project_name = projectName
+        if (siteId) params.site_id = siteId
```

Comment:
Funny how `params` is `const` but you can add stuff to it? Oh, funny javascript. :)

Reply:
But this would cause an error, it's the wild west of languages. 🤠 
```js
const params = {}

params = {}
```